core-bundle/contao/library/Contao/ArrayUtil.php

```diff
@@ -64,10 +64,14 @@ public static function recursiveKeySort(array &$array): void
 	 * @param mixed $arrArray
 	 *
 	 * @return boolean
+	 *
+	 * @deprecated Deprecated in Contao 5.4, to be removed in Contao 6. Use the array_is_list native PHP function instead.
 	 */
 	public static function isAssoc($arrArray): bool
 	{
-		return \is_array($arrArray) && !array_is_list($arrArray);
+		trigger_deprecation('contao/core-bundle', '5.4', 'Using "%s()" has been deprecated and will no longer work in Contao 6. Use the array_is_list native PHP function instead.', __METHOD__);
+
+		return !array_is_list($arrArray);
```

Comment:
You must not remove the `is_array()` check:

<img width="830" alt="" src="https://github.com/contao/contao/assets/1192057/ca5fd1b0-9dd5-4662-876b-bbcf53cd94c4">

Reply:
hmm, I somehow didn't see that. Kinda defeats the purpose of the deprecation? 🙈

Reply:
True, it might have some utility after all.